src/Microsoft.AspNetCore.Pubternal.Analyzers/PubternalityAnalyzer.cs

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.AspNetCore.Pubternal.Analyzers
```

Comment:
Should this just be `Internal.AspNetCore.Analyzers`? I'd much rather not make it specific to pubternal. It'll give us the option to add mroe stuff in the future

Reply:
I suggest `Microsoft.AspNetCore.Internal.Analyzers`. This is an analyzer that is internal to us, but otherwise general-purpose

Reply:
We have precedence for the other naming convention for our internal stuff e.g. `Internal.AspNetCore.Sdk`. Just reduces the risks of accidentally confusing it with a shipping package.